### PR TITLE
Add convert_file_to_pdf helper using libreoffice

### PR DESCRIPTION
**Pull request by baitsguy**
Tested with docx, but the library can support more. Haven't done perf testing, will be done next

*Note: Our docker installation currently doesn't install libreoffice, I will update it + docs we end up needing it*

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -65,6 +65,8 @@ jobs:
         run: sudo apt-get update
       - name: Install poppler and tesseract
         run: sudo apt-get install -y poppler-utils tesseract-ocr
+      - name: Install libreoffice
```
Comment (eric-anderson) [inline .github/workflows/testing.yml:68]: install it on the previous line. Change the name to "install apt dependencies"

File: lib/sycamore/sycamore/tests/unit/utils/test_fileformat_tools.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from io import BytesIO
+
+from pypdf import PdfReader
+
+import sycamore
+from sycamore.utils.fileformat_tools import convert_file_to_pdf
+from sycamore.tests.config import TEST_DIR
+
+
+def test_pdf_to_pdf():
```
Comment (eric-anderson) [inline lib/sycamore/sycamore/tests/unit/utils/test_fileformat_tools.py:13]: test_file_to_pdf?

File: lib/sycamore/sycamore/utils/fileformat_tools.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                "pdf",
+                source_path,
+                "--outdir",
+                output_path,
+            ]
+        )
+
+    assert doc.binary_representation is not None
+
+    current_filetype = doc.properties.get("filetype")
```
Comment (eric-anderson) [inline lib/sycamore/sycamore/utils/fileformat_tools.py:29]: .get("filetype", "unknown")

and remove the assertion. Unless libreoffice is using the extention for file type info

File: lib/sycamore/sycamore/utils/fileformat_tools.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    assert current_filetype is not None, "Document requires properties.filetype"
+
+    with NamedTemporaryFile(suffix=f".{current_filetype.split('/')[-1]}") as temp_file:
+        temp_file.write(doc.binary_representation)
+        temp_file.flush()
+
```
Comment (eric-anderson) [inline lib/sycamore/sycamore/utils/fileformat_tools.py:35]:
```
temp_path = Path(temp_file)
run_libreoffice(temp_file.name, temp_path.parent)
with open(temp_path.parent / temp_path.stem + ".pdf", "rb") as processed_file:
    doc.binary_representation = processed_file.read()
```

File: lib/sycamore/sycamore/utils/fileformat_tools.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import subprocess
+from tempfile import NamedTemporaryFile
+
+from sycamore.data import Document
+
+
+def convert_file_to_pdf(doc: Document) -> Document:
```
Comment (eric-anderson) [inline lib/sycamore/sycamore/utils/fileformat_tools.py:7]: binary_representation_to_pdf

I would expect convert_file_to_pdf(source_name, dest_name)

File: lib/sycamore/sycamore/utils/fileformat_tools.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        temp_file.write(doc.binary_representation)
+        temp_file.flush()
+
+        output_dir = temp_file.name.rsplit("/", 1)[0]
+        output_file_base = temp_file.name.rsplit(".", 1)[0]
+        run_libreoffice(temp_file.name, output_dir)
+
+        output_pdf_path = f"{output_file_base}.pdf"
+        with open(output_pdf_path, "rb") as processed_file:
+            doc.binary_representation = processed_file.read()
```
Comment (eric-anderson) [inline lib/sycamore/sycamore/utils/fileformat_tools.py:42]: doc.properties["filetype"] = "pdf"

File: lib/sycamore/sycamore/tests/unit/utils/test_fileformat_tools.py (hunk 2 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/tests/unit/utils/test_fileformat_tools.py:13]: oh yeah typo lol

File: lib/sycamore/sycamore/utils/fileformat_tools.py (hunk 3 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:29]: it does from what I see so gonna leave this

File: lib/sycamore/sycamore/utils/fileformat_tools.py (hunk 4 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:35]: the file extension when reading .docx was causing the Path lib to break (it includes multiple docs, e.g. 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'), so I needed to do it this way. 
I can choose to drop the file extensionbut figured its more useful when debugging

File: lib/sycamore/sycamore/utils/fileformat_tools.py (hunk 3 above)
Comment (baitsguy) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:29]: actually its a little unclear, their docs aren't very clear. Still useful for us to keep so we aren't supporting magic types

Comment (eric-anderson) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:29]: ok. If it's unclear what libreoffice will do it's safer to do the clearly correct thing.

Comment (baitsguy) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:29]: Which is enforcing/being aware of the extension I'd think?

Comment (eric-anderson) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:29]: Per discussion, get the extension from the original filesystem.

File: lib/sycamore/sycamore/utils/fileformat_tools.py (hunk 6 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/utils/fileformat_tools.py:42]: Per discussion this should have been application/pdf

**Review by eric-anderson — APPROVED** [review]

**Pull request merged**